src/main/java/com/exasol/adapter/installer/JarProvider.java

```diff
@@ -0,0 +1,13 @@
+package com.exasol.adapter.installer;
+
+/**
+ * This interface provides a {@link Jar}.
+ */
+public interface JarProvider {
+    /**
+     * Provide a {@link Jar}.
+     *
+     * @return new instance of {@link Jar}
+     */
+    Jar provideJar();
```

Comment:
I find the name a bit confusing, as it makes think this is a command method instead of a query method. Maybe we can rename it?
```suggestion
    Jar getJar();
```